### DISCUSSION TITLE
How can I get the "remote-outbound-rtp" using getStats()?

### DISCUSSION THREAD

**Discussion by bigbean17** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1859] [Q&A]
I'm trying to get the value of "remote-outbound-rtp" of a video stream. I found that "remote-outbound-rtp" only occur when it is the audio stream. I'm iterating the result of `RTCPeerConnection.getStatus(null)`, not sure what went wrong. Any help would be appreciated.

**Comment by niyatim23** [comment]
Hi @bigbean17, the rtp-outbound stats can be found below:
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/0ae9ec6b80deff9df724555927fb6595a23054f1/src/include/com/amazonaws/kinesis/video/webrtcclient/Stats.h#L345-L390

You can use the function below with the correct rtcMetrics type:
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/0ae9ec6b80deff9df724555927fb6595a23054f1/src/source/Metrics/Metrics.c#L204

An example of its usage is:
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/0ae9ec6b80deff9df724555927fb6595a23054f1/samples/Common.c#L135-L143

**Reply by bigbean17** [reply]
Hi @niyatim23, thanks for the solution, I saw the "remote-outbound-rtp" is empty, is that the reason why I cannot get the "remote-outbound-rtp" using `RTCPeerConnection.getStatus(null)` at frontend for video streaming? And I'm using the amazon-kinesis-video-streams-webrtc-sdk-js as the viewer at frontend. 
<img width="421" alt="image" src="https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/assets/69971426/9c8ec176-8b75-4726-bb8c-5a1622a7da12">

**Reply by niyatim23** [reply]
You mean you want to see video-related stats on the viewer side (the browser)? And you want to understand if we send them from the SDK or not?

**Reply by bigbean17** [reply]
> You mean you want to see video-related stats on the viewer side (the browser)? 


Yes
I'm trying to display the information in "REMOTE_OUTBOUND_RTP" but "REMOTE_OUTBOUND_RTP" doesn't present in the result of `getStats()`.

**Reply by disa6302** [reply]
[REMOTE_OUTBOUND_RTP](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/src/source/Metrics/Metrics.c#L244) is unsupported at the moment. If you would like to view `REMOTE_OUTBOUND_RTP`, you need to follow the instructions @niyatim23  mentioned. Our samples have that example. `getStats()` is not a WebRTC C SDK API and you need to use `rtcPeerConnectionGetMetrics ` and pass in the necessary parameters to get the stats.

**Reply by bigbean17** [reply]
Ok I will have a try, thanks~
